src/App.tsx

```diff
@@ -1,14 +1,60 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+// import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
+
+export enum Filters {
+  all = 'all',
+  completed = 'completed',
+  active = 'active',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [, setError] = useState(null);
+  const [showDetails, setShowDetails] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Filters>(Filters.all);
+  const [query, setQuery] = useState<string>('');
```

Comment:
type here will be inferred from default value 
```suggestion
  const [query, setQuery] = useState('');
```